mkvtoolnix-app 29.0.0 (new cask)

MikeMcQuaid opened the pull request:
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [ ] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

This cask has been previously refused due to duplicating the Homebrew formula but the Homebrew formula will be GUI-less when https://github.com/Homebrew/homebrew-core/pull/35383 is merged and this provides the GUI and a macOS .app.

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

MikeMcQuaid commented:
Build failure is expected due to having the same name as the Homebrew/core formula. I believe it's against the guidelines but a `mkvtoolnix-app` or `-gui` or something would be possible but I'd probably favour just keeping the names identical.

Note this may warrant some future policy/documentation changes as other GUIs we remove from Homebrew/core may find themselves a better fit for Homebrew/cask.

vitorgalvao commented on Casks/mkvtoolnix.rb:
```diff
@@ -0,0 +1,11 @@
+  name 'MKVToolNix'
+  homepage 'https://mkvtoolnix.download/'
+
+  app "MKVToolNix-#{version}.app"
```
Removed `target` because we tend to keep the developer’s name, meaning we [don’t remove versions from the name](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/app.md#target-should-only-be-used-in-select-cases).

vitorgalvao commented:
Hope you don’t mind me editing the PR directly. We don’t shy from that in this repo, and always squash and merge.

---

An [`-app` suffix is foreseen in the rules](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md#token-overlap-with-a-formula) in a specific case. Does this fit? Is the `app` “simply a wrapper of the CLI tool”?

MikeMcQuaid commented:
> Ready to merge unless `-app` is warranted (i.e. if the GUI is nothing more than a CLI wrapper).

@vitorgalvao I think that applies here, yes 👍. Renamed, squashed, pushed.

> Hope you don’t mind me editing the PR directly. We don’t shy from that in this repo, and always squash and merge.

Always fine with me 👍

vitorgalvao commented:
Thank you!
